lua/orgmode/agenda/init.lua

```diff
@@ -25,36 +25,53 @@ function Agenda:new(opts)
     views = {},
     content = {},
     highlights = {},
+    win_width = utils.winwidth(),
   }
   setmetatable(data, self)
   self.__index = self
   return data
 end
 
 function Agenda:agenda(opts)
+  self:open_window()
+  self.win_width = utils.winwidth()
   local view = AgendaView:new(vim.tbl_deep_extend('force', opts or {}, {
     filters = self.filters,
+    win_width = self.win_width,
   })):build()
   self.views = { view }
   return self:_render()
 end
 
 -- TODO: Introduce searching ALL/DONE
 function Agenda:todos()
-  local view = AgendaTodosView:new({ filters = self.filters }):build()
+  self:open_window()
+  self.win_width = utils.winwidth()
```

Comment:
I think we can move this into the `open_window` method, so it doesn't have to be set from each agenda view type, to avoid some duplication.
Another solution might be to have like a generic method that will accept the "View" class, and you just initialize what's passed in, since all of the views are initialized with the same parameters. Something like `self:open_agenda_view(AgendaView)`

Reply:
I went with the latter option, as it trims of quite some lines